### PR TITLE
ODPM-541: core logging improvement: error-log name of a plugin not initialized

### PR DESCRIPTION
**Pull request by VladoLavor**
Signed-off-by: Vladimir Lavor <vlavor@cisco.com>

### CODE REVIEW

File: core/agent_core.go [hunk 1]
```diff
@@ -146,23 +158,27 @@ func (agent *Agent) initPlugins() error {
...
 		if plug2, ok := plug.Plugin.(PostInit); ok {
 			agent.Debug("afterInit begin for ", plug.PluginName)
 			err := plug2.AfterInit()
 			if err != nil {
 				agent.Stop()
-
 				return fmt.Errorf(logPostErrorFmt, plug.PluginName, err)
 			}
 			agent.Info(fmt.Sprintf(logPostSuccessFmt, plug.PluginName))
+			agent.loadedPlugins = append(agent.loadedPlugins, plug)
```
Comment (jozef-slezak) [inline core/agent_core.go]: this needs to be a different collection (please figure out also proper name for init() collection myabe just suffix init)

File: core/agent_core.go [hunk 2]
```diff
@@ -32,26 +32,35 @@ var (
...
+	startup
+}
+
+type startup struct {
+	// The startup/initialization must take no longer that maxStartup.
+	MaxStartupTime time.Duration
+	// loaded plugins
+	loadedPlugins []*NamedPlugin
+	// the field is set before initialization of every plugin with its name
+	handledPlugin string
```
Comment (jozef-slezak) [inline core/agent_core.go]: currentlyProcessingPlug (better name?)

File: core/agent_core.go [hunk 3]
```diff
@@ -97,7 +106,8 @@ func (agent *Agent) Start() error {
 		return nil
 	case <-time.After(agent.MaxStartupTime):
 		//TODO FIX - stop the initialization and close already initialized
-		return fmt.Errorf("%s", "Some plugins not intialized before timeout")
+		return fmt.Errorf("plugin %v not completed before timeout. Initialized %d plugins of %d: %v",
+			agent.handledPlugin, len(agent.loadedPlugins), len(agent.plugins), agent.loadedPlugins)
```
Comment (jozef-slezak) [inline core/agent_core.go]: Just log using multiple lines the ok ones and the problematic ones

Comment (VladoLavor) [reply core/agent_core.go]: sample log:
ERRO[0015] Error loading core: plugin DatasyncExample Init() not completed before timeout.
Init succ: [ServiceLabel StatusCheck ResyncOrch ETCD ETCDDataSync] 
Init err: [DatasyncExample] 
AfterInit succ: [] 
AfterInit err: [ServiceLabel StatusCheck ResyncOrch ETCD ETCDDataSync DatasyncExample]  loc="core/event_loop.go(27)" tag=00000000

**Pull request merged**